rest/route/service.go

```diff
@@ -161,6 +161,7 @@ func AttachHandler(app *gimlet.APIApp, opts HandlerOpts) {
 	app.AddRoute("/projects/{project_id}/tag/{tag}").Version(2).Get().Wrap(checkUser, checkProjectAdmin).RouteHandler(makeCheckProjectTag(sc))
 	app.AddRoute("/projects/{project_id}/tag/{tag}").Version(2).Post().Wrap(checkUser, checkProjectAdmin, editProjectSettings).RouteHandler(makeAddProjectTag(sc))
 	app.AddRoute("/permissions").Version(2).Get().RouteHandler(&permissionsGetHandler{})
+	app.AddRoute("/repos/{repo_id}").Version(2).Get().Wrap(checkUser, checkProjectAdmin, viewProjectSettings).RouteHandler(makeGetRepoByID(sc))
```

Comment:
So I actually think there isn't an authentication that fits this: it's a bit complicated, but checkProjectAdmin checks explicitly that ProjectRef is set when loading the context, and it's not because we use repo_id. I think general repo authentication might be better as a separate ticket / is out of scope of this ticket, but for now I would say that "viewProjectSettings" is enough, and you can leave out checkProjectAdmin. (We can talk about this more in 1:1 if you'd like)

Reply:
https://jira.mongodb.org/browse/EVG-13671